components/TodoList.js

```diff
@@ -105,55 +88,69 @@ function TodoList() {
         id: Math.random().toString(32).substring(2),
         text: newTodo,
         priority: priority,
+        completed: false,
       };
       setTodos([...todos, todoItem]);
       setNewTodo('');
     }
   };
 
   const handleRemoveTodo = (id) => {
-    const confirmDelete = window.confirm('Are you sure you want to delete this task?');
+    const confirmDelete = window.confirm('Bist du sicher dass du es löschen möchtest ');
     if (confirmDelete) {
       const newTodos = todos.filter((todo) => todo.id !== id);
       setTodos(newTodos);
     }
   };
 
   const handleAddNote = (id, note) => {
-    const noteWords = note.split(' ');
-
-    if (noteWords.length <= 1 && note.length <= 12) {
-      const truncatedNote = noteWords.slice(0, 6).join(' ');
-      setNotes((prevNotes) => ({ ...prevNotes, [id]: truncatedNote }));
+    if (note && note.trim() !== '') { 
+      const noteWords = note.split(' ');
+  
+      if (noteWords.length <= 1 && note.length <= 12) {
+        const truncatedNote = noteWords.slice(0, 6).join(' ');
+        setNotes((prevNotes) => ({ ...prevNotes, [id]: truncatedNote }));
+      } else {
+        alert('Maximal 12 Zeichen sind für Notizen erlaubt.');
+      }
     } else {
-      alert('Maximal 12 Zeichen sind für Notizen erlaubt.');
+      alert('Bitte geben Sie eine gültige Notiz ein.');
     }
   };
 
+  const handleTaskCompletion = (taskId) => {
+    const updatedTodos = todos.map((todo) =>
+      todo.id === taskId ? { ...todo, completed: true } : todo
+    );
+
+    setTodos(updatedTodos);
+  };
+
   const priorityGroups = {
     high: todos.filter((todo) => todo.priority === 'high'),
     medium: todos.filter((todo) => todo.priority === 'medium'),
     low: todos.filter((todo) => todo.priority === 'low'),
   };
 
   return (
-    <div>
-      <h2>Tasks</h2>
-      <input type="text" value={newTodo} onChange={handleNewTodoChange} />
-      <select value={priority} onChange={handlePriorityChange}>
-        <option value="high">High Priority</option>
-        <option value="medium">Medium Priority</option>
-        <option value="low">Low Priority</option>
-      </select>
-      <button onClick={handleAddNewTodo}>Add</button>
-
-      <FlexContainer>
+    <CenteredContainer>
+      <div>
```

Comment:
i think you don't need a div here, because the centeredContainer is already a div